pkg/controllers/state/cluster.go

```diff
@@ -68,6 +70,48 @@ func NewCluster(clk clock.Clock, client client.Client, cp cloudprovider.CloudPro
 	}
 }
 
+// Synced validates that the Machines and the Nodes that are stored in the apiserver
+// have the same representation in the cluster state. This is to ensure that our view
+// of the cluster is as close to correct as it can be when we begin to perform operations
+// utilizing the cluster state as our source of truth
+func (c *Cluster) Synced(ctx context.Context) bool {
+	machineList := &v1alpha5.MachineList{}
+	if err := c.kubeClient.List(ctx, machineList); err != nil {
+		// If it's a KindMatch error, this just means the Machine CRD isn't registered
+		// TODO @joinnis: Remove this once Machines are part of the Karpenter chart
+		if !meta.IsNoMatchError(err) {
+			return false
+		}
+	}
+	nodeList := &v1.NodeList{}
+	if err := c.kubeClient.List(ctx, nodeList); err != nil {
+		return false
```

Comment:
Same as above, just because synced() is normally called in a loop (or was). If something goes wrong, we want to know why vs. just hanging.